kore/src/Kore/Builtin/Bytes.hs

```diff
@@ -0,0 +1,371 @@
+import qualified Data.Text.Encoding as E
+
+import qualified Kore.Attribute.Pattern as Attribute
+import qualified Kore.Builtin.Builtin as Builtin
+import Kore.Builtin.Bytes.Bytes
+import qualified Kore.Builtin.Int as Int
+import qualified Kore.Builtin.String as String
+import qualified Kore.Internal.Pattern as Pattern
+import Kore.Internal.TermLike
+import Kore.Step.Simplification.Simplify
+
+-- | Verify that the sort is hooked to the @Bytes@ sort.
+-- | See also: 'sort', 'Builtin.verifySort'.
+assertSort :: Builtin.SortVerifier
+assertSort = Builtin.verifySort sort
+
+-- | Verify that hooked sort declarations are well-formed.
+-- | See also: 'Builtin.verifySortDecl'.
+sortDeclVerifiers :: Builtin.SortDeclVerifiers
+sortDeclVerifiers = HashMap.fromList [ (sort, Builtin.verifySortDecl) ]
+
+-- | Verify that hooked symbol declarations are well-formed.
+-- | See also: 'Builtin.verifySymbol'.
+symbolVerifiers :: Builtin.SymbolVerifiers
+symbolVerifiers =
+    HashMap.fromList
+    [   ( bytes2StringKey
+        , Builtin.verifySymbol string [bytes]
+        )
+    ,   ( string2BytesKey
+        , Builtin.verifySymbol bytes [string]
+        )
+    ,   ( updateKey
+        , Builtin.verifySymbol bytes [bytes, int, int]
+        )
+    ,   ( getKey
+        , Builtin.verifySymbol int [bytes, int]
+        )
+    ,   ( substrKey
+        , Builtin.verifySymbol bytes [bytes, int, int]
+        )
+    ,   ( replaceAtKey
+        , Builtin.verifySymbol bytes [bytes, int, bytes]
+        )
+    ,   ( padRightKey
+        , Builtin.verifySymbol bytes [bytes, int, int]
+        )
+    ,   ( padLeftKey
+        , Builtin.verifySymbol bytes [bytes, int, int]
+        )
+    ,   ( reverseKey
+        , Builtin.verifySymbol bytes [bytes]
+        )
+    ,   ( lengthKey
+        , Builtin.verifySymbol int [bytes]
+        )
+    ,   ( concatKey
+        , Builtin.verifySymbol bytes [bytes, bytes]
+        )
+    ]
+  where
+    bytes  = assertSort
+    int    = Int.assertSort
+    string = String.assertSort
+
+expectBuiltinBytes
+    :: MonadSimplify m
+    => Text
+    -> TermLike variable
+    -> MaybeT m (Symbol, ByteString)
+expectBuiltinBytes ctx =
+    \case
+        Bytes_ _ symbol bytesValue -> return (symbol, bytesValue)
+        _ ->
+            Builtin.verifierBug
+            $ Text.unpack ctx ++ ": Term not a bytes value"
+
+evalBytes2String :: Builtin.Function
+evalBytes2String =
+    Builtin.functionEvaluator evalBytes2String0
+  where
+    evalBytes2String0 :: Builtin.FunctionImplementation
+    evalBytes2String0 resultSort = \arguments ->
+        Builtin.getAttemptedAxiom $ do
+            let _bytes =
+                    case arguments of
+                        [_bytes] -> _bytes
+                        _ -> Builtin.wrongArity bytes2StringKey
+            (_, bytestring) <- expectBuiltinBytes bytes2StringKey _bytes
+            Builtin.appliedFunction
+                . String.asPattern resultSort
+                . E.decodeUtf8
```

Comment:
We may need to ask @dwightguth, but I think that a raw 8-bit encoding is expected, not UTF-8.